modules/programs/vim.nix

```diff
@@ -55,6 +55,24 @@ let
     in
       optionalString (value != null) ("set " + v);
 
+  plugins = flip map cfg.plugins (plugin:
+    if builtins.isString plugin
+    then warn
+      "Vim plugins should be added as packages instead of strings. Please update your configuration."
+      (
+        findFirst
```

Comment:
Wouldn't it be sufficient to look at the attribute name in `pkgs.vimPlugins`? Then could do something like

```
plugins =
  let
    vpkgs = pkgs.vimPlugins;
    getPkg = v:
      if isDerivation v then [ v ]
      else optional (isString v && hasAttr v vpkgs) vpkgs.${v};
  in
    concatMap getPkg cfg.plugins;
```
which I think is more straight-forward.